### PR TITLE
[DES-278] Update multi section assignment dialog to match redlines

### PR DESCRIPTION
**Pull request by levadadenys**
### [DES-278] Update multi section assignment dialog to match redlines

* use Typography and Checkbox components from design system
* updated styles to match [readlines](https://www.figma.com/file/fe70qoiGWnV78j9MKfAEdo/Teacher-Experience-UX%2FUI-Improvements?type=design&node-id=3733%3A36101&t=E9LYVUFPqfIgRTpk-1)
* removed TeacherSectionOption since we don't really need it now
* updated tests

Before:
Знімок екрана 2023-07-03 о 21 24 21

After:
Знімок екрана 2023-07-06 о 16 36 43

## Links
- jira ticket: [DES-278](https://codedotorg.atlassian.net/browse/DES-278)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by levadadenys** [comment]
PR is 99% done, just waiting for Typography update to be merged so I can also use body-two text for modal description.
Meaning PR can be reviewed already.

**Comment by levadadenys** [comment]
@megcrenshaw @molly-moen `apps/src/templates/AccessibleDialog.jsx` has a different size and padding then it's specified in the readlines. I believe the on @megcrenshaw implemented is better since it's responsive, but also I believe that that's something it would be better if both of you discuss (because @megcrenshaw  is the author of that component)

File: apps/src/templates/MultipleSectionsAssigner.jsx [hunk 1]
```diff
@@ -232,57 +250,6 @@ MultipleSectionsAssigner.propTypes = {
...
-    fontFamily: '"Gotham 5r", sans-serif',
-  },
-  selectAllSectionsLabel: {
-    fontFamily: "'Gotham 5r', sans-serif",
-    fontSize: 16,
-    cursor: 'pointer',
-    color: color.link_color,
-  },
-};
-
```
Comment (megcrenshaw) [inline apps/src/templates/MultipleSectionsAssigner.jsx:313]: Wahoo so excited to see this gone!

File: apps/src/templates/MultipleSectionsAssigner.jsx [hunk 2]
```diff
@@ -162,52 +167,65 @@ const MultipleSectionsAssigner = ({
...
+          moduleStyle.sectionList
+        )}
+      >
+        <Heading5>{i18n.yourSectionsList()}</Heading5>
+        <div className={moduleStyle.sectionListOptionsContainer}>
+          {sections &&
+            sections.map(
+              section =>
+                isAssignableToSection(section.participantType) && (
+                  <Checkbox
```
Comment (megcrenshaw) [inline apps/src/templates/MultipleSectionsAssigner.jsx:222]: Love this refactor 🙌

**Review by sanchitmalhotra126 — APPROVED** [review]
Don't have a ton of context on this feature, but the changes look good to me!

**Review by megcrenshaw — APPROVED** [review]
Manually verified that the selections are working locally. This looks wonderful! Thanks for your work on this, Denys!

**Comment by levadadenys** [comment]
Updated description section to be bodyTwoText.
Waiting for drone build to pass and then'll be ready to merge this PR.

**Pull request merged**